### DISCUSSION TITLE
[QUESTION] Integration of GeoSPARQL into OML project

### DISCUSSION THREAD

**Discussion by husseinmarah** [opencaesar/platform#27] [Q&A]
Hi everyone,

Lately, I've been trying to use GeoSPARQL standard: [https://docs.ogc.org/is/22-047r1/22-047r1.html](GeoSPARQL ). The issue is that when I import the ontology into the project, it says that the namespace cannot be resolved. The error message: "Namespace <http://www.opengis.net/ont/geosparql#> could not be resolved to a vocabulary".
Does anyone have an idea about how to use GeoSPARQL or other spatial ontology inside an OML project?

**Comment by melaasar** [comment]
Hi. Sorry for the delay.  Did you first convert the ontology to OML? How did you import it into the project? Please provide details.

**Reply by husseinmarah** [reply]
Hi Maged,

Thanks for your reply. Actually I tried different approaches, first I tried to convert the ontology which is in ttl to oml with the owl2oml, but the tool did not work.
Also, I tried to import the ontology directly to my project (oml model), but then I figured out that without converting the ontology to oml, this will not work.
So, do you have any suggestions to make it work?

Best,
Hussein

**Reply by melaasar** [reply]
Ah, currently the OWL2OML adapter assumes the OWL is produced from OML by running OML2OWL. However, the adapter's repo has support for converting OWL that was not produced by OML but this is currently not a released capability. If you like to run this, you can clone the repo (https://github.com/opencaesar/owl-adapter) and change this line of code from Owl2Oml to Owl2OmlEx.

https://github.com/opencaesar/owl-adapter/blob/6834036aa19b6f77d8f451cc30aeb9da86df8ac6/owl2oml/src/main/java/io/opencaesar/owl2oml/Owl2OmlApp.java#L251

Also, notice that to run this transformation properly, copy the owl ontologies under src/owl in a properly setup OML project (like what you get from creating a new OML project in OML Rosetta, or if you cloned [oml-template](https://github.com/opencaesar/oml-template/) repo).

If you like to see this capability part of the release, please upvote this issue.

**Comment by husseinmarah** [comment]
Thanks for the answer.

I will try to use the adapter that you mentioned.
I have another question on how to add a dependency to a OML project compiled by Rosseta. I tried to add the GeoSPARQL dependency in the build.gradle of my project in "The Gradle task dependencies"  section as:
`classpath 'org.apache.jena:jena-geosparql:5.2.0'`

also, I added the dependency in the "The OML dependencies" section as the following:
`implementation 'org.apache.jena:jena-geosparql:5.2.0'`

But the dependency is still not compiled with the project!
